src/drivers/fluid_portaudio.c

```diff
@@ -61,10 +61,67 @@ void delete_fluid_portaudio_driver (fluid_audio_driver_t *p);
 
 #define PORTAUDIO_DEFAULT_DEVICE "PortAudio Default"
 
+/*
+ * Checks if num_device number is a valid device and returns the name of the
+ * portaudio device.
+ * A device is valid if it is an output device with more than 2 channels.
+ * 
+ * @param num_device: index of the portaudio device to check.
+ * @name_ptr: address of the pointer on returned name,if num_device is valid
+ * otherwise the name is not returned.
+ *
+ * The name returned is unique for each num_device index, so this 
+ * name is useful to identify any available host audio device.
+ * This name is convenient for audio.portaudio.device setting.
+ *
+ * The format of the name is: device_index:host_api_name:host_device_name
+ *   
+ *   example: 5:MME:SB PCI
+ *   
+ *   5: is the portaudio device index.
+ *   MME: is the host API name.
+ *   SB PCI: is the host device name.
+ * 
+ * @return: TRUE if device_num is a valid output device, FALSE otherwise.
+ * When TRUE, the name is returned in allocated memory. The caller must check
+ * the name pointer for a valid memory allocation and should free the memory.
+ */
+char get_valid_portaudio_device_name(int num_device, char **name_ptr)
```

Comment:
>It should be static
Ok.

The function is intended to return 3 informations:
1) true/false to say that the device is  valid/invalid.
2.1) When the device is valid, the name is returned or NULL if error of allocation.
2.2) When the device is invalid. This is not an error. The name has no meaning.
Anyway, the return value should be a int rather than char.

Reply:
Ok, but then please return `FLUID_OK` or `FLUID_FAILED`.

Reply:
Ok.